server/api/rule.go

```diff
@@ -42,6 +43,42 @@ func newRulesHandler(svr *server.Server, rd *render.Render) *ruleHandler {
 	}
 }
 
+type ruleMiddleware struct {
+	s  *server.Server
+	rd *render.Render
+	*server.Handler
+}
+
+func newRuleMiddleware(s *server.Server, rd *render.Render) ruleMiddleware {
```

Comment:
Shall we move it to middleware.go?

Reply:
Because it is only used in rulesHander, I put it in rules.go.

Reply:
Both are ok to me.